src/Domain/FixerFileProcessor.php

```diff
@@ -0,0 +1,190 @@
+     */
+    private $errorAndDiffCollector;
+    /**
+     * @var \Symplify\EasyCodingStandard\FixerRunner\Parser\FileToTokensParser
+     */
+    private $fileToTokensParser;
+    /**
+     * @var \NunoMaduro\PhpInsights\Domain\Differ
+     */
+    private $differ;
+
+    public function __construct(
+        CachedFileLoader $cachedFileLoader,
+        ErrorAndDiffCollector $errorAndDiffCollector,
+        FileToTokensParser $fileToTokensParser,
+        Differ $differ
+    ) {
+        $this->cachedFileLoader = $cachedFileLoader;
+        $this->errorAndDiffCollector = $errorAndDiffCollector;
+        $this->fileToTokensParser = $fileToTokensParser;
+        $this->differ = $differ;
+    }
+
+    public function addChecker(FixerInterface $fixer): void
+    {
+        $this->fixers[] = $fixer;
+    }
+
+    /**
+     * @return array<\PhpCsFixer\Fixer\FixerInterface>
+     */
+    public function getCheckers(): array
+    {
+        return $this->fixers;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function processFile(SmartFileInfo $smartFileInfo): string
+    {
+        $filePath = $smartFileInfo->getRealPath();
+        if ($filePath === false) {
+            throw new \LogicException('Unable to found file ' . $smartFileInfo->getFilename());
+        }
+        $oldContent = $this->cachedFileLoader->getFileContent($smartFileInfo);
+        $tokens = $this->fileToTokensParser->parseFromFilePath($filePath);
+
+        /** @var FixerInterface $fixer */
+        foreach ($this->getCheckers() as $fixer) {
+            $this->processTokensByFixer($smartFileInfo, $tokens, $fixer);
+            if (! $tokens->isChanged()) {
+                continue;
+            }
+
+            $newContent = $tokens->generateCode();
+            $fullDiff = $this->differ->diff($oldContent, $newContent);
+
+            $this->processDiff($fullDiff, get_class($fixer), $smartFileInfo);
+
+            Tokens::clearCache();
+            // Reinit tokens
+            $tokens = $this->fileToTokensParser->parseFromFilePath($filePath);
+        }
+
+        return 'processed';
+    }
+
+    private function processTokensByFixer(SmartFileInfo $smartFileInfo, Tokens $tokens, FixerInterface $fixer): void
+    {
+        try {
+            $fixer->fix($smartFileInfo, $tokens);
+        } catch (\Throwable $throwable) {
+            throw new FixerFailedException(sprintf(
+                'Fixing of "%s" file by "%s" failed: %s in file %s on line %d',
+                $smartFileInfo->getRelativeFilePath(),
+                get_class($fixer),
+                $throwable->getMessage(),
+                $throwable->getFile(),
+                $throwable->getLine()
+            ), $throwable->getCode(), $throwable);
+        }
+
+        if (! $tokens->isChanged()) {
+            return;
+        }
+    }
+
+    private function processDiff(string $diff, string $fixerClass, SmartFileInfo $smartFileInfo): void
```

Comment:
So basically you have to analyse a diff file to actually split up the content?

Reply:
Get what you mean that some refactoring is required when the `Details` class is merged in. 
When you get to that part, I think we should have a new param on Details class for the complete diff file. 
Then the formatters can do most of the heavy work of analysing the diff.

Reply:
You are right ! It's probably more the Formatter role :wink: I will move this parse in CSFixer Insight, and use the errorAndDiffCollector to store the full diff.